src/helpers/fetchData.js

```diff
@@ -0,0 +1,36 @@
+const fetch = require('node-fetch');
+const { YOUTUBE_API_KEY, YOUTUBE_CHANNEL_ID } = require('../config');
+
+async function fetchLatestYoutubeVideos({ maxResults, publishedAfter }) {
+  const mrOpt = maxResults ? `maxResults=${maxResults}` : '';
+  const paOpt = publishedAfter ? `publishedAfter=${publishedAfter}` : '';
+  const optUrl = [mrOpt, paOpt].filter(Boolean).join('&') || '';
+  const apiUrl = `https://www.googleapis.com/youtube/v3/search?key=${YOUTUBE_API_KEY}&channelId=${YOUTUBE_CHANNEL_ID}&part=snippet&order=date&${optUrl}`;
+  try {
+    const resp = await fetch(apiUrl);
+    const { items, error } = await resp.json();
+
+    if (error && error.errors[0].reason === 'keyInvalid') {
+      // eslint-disable-next-line no-console
+      console.log(`Invalid 'YOUTUBE_API_KEY'`);
+      return [];
+    }
+
+    return items.map(({ id: { videoId: videoID }, snippet: { title: name, publishedAt: date, description, thumbnails: { high: { url: thumbnail } } } }) => {
```

Comment:
I just tested this locally and I got `TypeError: Cannot read property 'map' of undefined`. We should account for the case of items not being in the response...?

Reply:
It might help to check this property `pageInfo.totalResults` and return if it is 0.

Reply:
@spiray 

i didn´t get that error msg after I checked if the `YOUTUBE_API_KEY` was set or not. This left me thinking. 

Maybe `
if (error && error.errors[0].reason === 'keyInvalid') {
      console.log('Invalid "YOUTUBE_API_KEY"');
      return [];
    }
` isn´t working as it should?

Are  you getting the `Invalid 'YOUTUBE_API_KEY'` log?

Gonna test it in a clean pull for the 5th time, lol.

Reply:
Tested and getting `Invalid 'YOUTUBE_API_KEY'`

 @spiray Can you console.log the `api Url` declared in line 8 and post it here. Maybe checking the raw response of the youtube-api give as some clue of what´s happening.

Reply:
`keyInvalid` isn't the only error that can occur. The first time I ran it, I got a `keyExpired` error even though the key was newly generated. Instead of checking for a specific error, just check if there is an error at all.

`if (error) log(error) return false/null`

It's also smart to check if the response is empty, if it is, return before it tries to map over it.